onedal/neighbors/neighbors.py

```diff
@@ -0,0 +1,369 @@
+                "enter integer value" %
+                type(n_neighbors))
+
+    if X is not None:
+        query_is_train = False
+        X = _check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32])
+    else:
+        query_is_train = True
+        X = estimator._fit_X
+        # Include an extra neighbor to account for the sample itself being
+        # returned, which is removed later
+        n_neighbors += 1
+
+    n_samples_fit = estimator.n_samples_fit_
+    if n_neighbors > n_samples_fit:
+        raise ValueError(
+            "Expected n_neighbors <= n_samples, "
+            " but n_samples = %d, n_neighbors = %d" %
+            (n_samples_fit, n_neighbors)
+        )
+
+    chunked_results = None
+
+    method = parse_auto_method(
+        estimator, estimator._fit_method, estimator.n_samples_fit_, n_features)
+
+    params = estimator._get_onedal_params(X)
+    prediction_results = _onedal_predict(estimator, estimator._onedal_model, X, params)
+
+    distances = backend.to_numpy(prediction_results.distances)
+    indices = backend.to_numpy(prediction_results.indices)
+
+    if method == 'kd_tree':
+        for i in range(distances.shape[0]):
+            seq = distances[i].argsort()
+            indices[i] = indices[i][seq]
+            distances[i] = distances[i][seq]
+
+    if return_distance:
+        results = distances, indices.astype(int)
+    else:
+        results = indices.astype(int)
+
+    if chunked_results is not None:
+        if return_distance:
+            neigh_dist, neigh_ind = zip(*chunked_results)
+            results = np.vstack(neigh_dist), np.vstack(neigh_ind)
+        else:
+            results = np.vstack(chunked_results)
+
+    if not query_is_train:
+        return results
+    # If the query data is the same as the indexed data, we would like
+    # to ignore the first nearest neighbor of every sample, i.e
+    # the sample itself.
+    if return_distance:
+        neigh_dist, neigh_ind = results
+    else:
+        neigh_ind = results
+
+    n_queries, _ = X.shape
+    sample_range = np.arange(n_queries)[:, None]
+    sample_mask = neigh_ind != sample_range
+
+    # Corner case: When the number of duplicates are more
+    # than the number of neighbors, the first NN will not
+    # be the sample, but a duplicate.
+    # In that case mask the first duplicate.
+    dup_gr_nbrs = np.all(sample_mask, axis=1)
+    sample_mask[:, 0][dup_gr_nbrs] = False
+    neigh_ind = np.reshape(
+        neigh_ind[sample_mask], (n_queries, n_neighbors - 1))
+
+    if return_distance:
+        neigh_dist = np.reshape(
+            neigh_dist[sample_mask], (n_queries, n_neighbors - 1))
+        return neigh_dist, neigh_ind
+    return neigh_ind
+
+class NeighborsBase(BaseNeighborsBase, metaclass=ABCMeta):
+    # @abstractmethod
+    def __init__(self, n_neighbors=None, radius=None,
+                 algorithm='auto', leaf_size=30, metric='minkowski',
+                 p=2, metric_params=None, n_jobs=None):

```

Comment:
We don't support n_job, leaf_size parameters for oneDAL KNN. Why do you add them?